src/services/ApiKeyService.ts

```diff
@@ -0,0 +1,34 @@
+import sha256 from "crypto-js/sha256";
+import ApiKeyRepository from "../repositories/ApiKeyRepository";
+import { nanoid } from "nanoid";
+
+export class ApiKeyService {
+  public async addApiKey(name: string, createdBy: string) {
+    const key = this.generateApiKey();
+
+    if (this.checkIfApiKeyExists(key)) this.addApiKey(name, createdBy);
+
+    const keyHash = sha256(key).toString();
+    const createdAt = new Date();
+
+    await ApiKeyRepository.addApiKey(name, keyHash, createdBy, createdAt);
+    return key;
+  }
+
+  public deleteApiKey = ApiKeyRepository.deleteApiKey;
+
+  public generateApiKey() {
```

Comment:
I think exporting this function could lead to confusion. See [TokenService#generateUnformattedToken](https://github.com/LeoTuet/backend-we-want-to-sleep/blob/main/src/services/TokenService.ts#L4) and call it generateUnformattedApiKey?